### PR TITLE
add Palette Button component

### PR DESCRIPTION
**Pull request by SilentTiger**
Please makes sure these boxes are checked before submitting your PR, thank you!

* [y] Make sure you follow the contributing guide.
* [y] Rebase before creating a PR to keep commit history clear.
* [y] Add some descriptions and refer relative issues for you PR.

this is a button set that has a cool animation, see this:
[](http://p1.bqimg.com/4851/207bb465d01e0736.gif)

### CODE REVIEW

File: packages/palette-button/src/palette-button.vue [hunk 1]
```diff
@@ -0,0 +1,146 @@
+<template>
+	<div class="palette-button" :class="{ expand: expanded, 'palette-button-active': transforming }" @animationend="onMainAnimationEnd">
```
Comment (QingWei-Li) [inline packages/palette-button/src/palette-button.vue]: 两个空格

Comment (QingWei-Li) [inline packages/palette-button/src/palette-button.vue]: animationend 事件是兼容 webkitAnimationEnd 的么，话说还需要前缀的么 @Leopoldthecoder

File: packages/palette-button/src/palette-button.vue [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+      this.slotChildren = [];
+      for (let i = 0; i < this.$slots.default.length; i++) {
+        if (this.$slots.default[i].elm.nodeType !== 3) {
+          this.slotChildren.push(this.$slots.default[i]);
+        }
+      }
+
+      let css = '';
+      let direction_arc = Math.PI * (3 + Math.max(['lt', 't', 'rt', 'r', 'rb', 'b', 'lb', 'l'].indexOf(this.direction), 0)) / 4;
+      console.log(this.direction, direction_arc, this);
```
Comment (QingWei-Li) [inline packages/palette-button/src/palette-button.vue]: console.log

File: packages/palette-button/src/palette-button.vue [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    destroyed() {
+      if (this.styleNode) {
+        this.styleNode.parentNode.removeChild(this.styleNode);
+      }
+    }
+  };
+</script>
+
+<style scoped>
+  .palette-button{
```
Comment (QingWei-Li) [inline packages/palette-button/src/palette-button.vue]: 不要用 scoped，不然用户无法覆盖样式，我们使用 BEM 的命名风格，统一用 .mint- 开头

File: packages/palette-button/src/palette-button.vue [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+  .sub-button-container>*{
+    position:absolute;
+    top:15px;
+    left:15px;
+    width:25px;
+    height:25px;
+    transition: transform .3s ease-in-out;
+  }
+
+  @keyframes zoom{
```
Comment (QingWei-Li) [inline packages/palette-button/src/palette-button.vue]: 动画名也加上前缀，避免重复

**Comment by QingWei-Li** [comment]
还有 commit rebase 一下，不要用中文

File: packages/palette-button/src/palette-button.vue (hunk 1 above)
Comment (Leopoldthecoder) [inline packages/palette-button/src/palette-button.vue]: http://caniuse.com/#search=animationend
看这里觉得安卓浏览器上可能会有问题。@SilentTiger 可以试一下。

**Pull request closed without merging**